### ISSUE TITLE
[Bug]: More than 1000 expressions in a list are not allowed on Oracle

### ISSUE THREAD

**Issue by ostasevych** [nextcloud/server#41605]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Hi! I've found in my logs the error entry:
```
[core] Error: Doctrine\DBAL\Query\QueryException: More than 1000 expressions in a list are not allowed on Oracle. at <<closure>>

0. /var/www/html/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php line 326
   OC\DB\QueryBuilder\QueryBuilder->execute()
1. /var/www/html/nextcloud/apps/activity/lib/Data.php line 513
   OC\DB\QueryBuilder\QueryBuilder->executeStatement()
2. /var/www/html/nextcloud/apps/activity/lib/Data.php line 380
   OCA\Activity\Data->deleteActivitiesForMySQL()
3. /var/www/html/nextcloud/apps/activity/lib/Data.php line 364
   OCA\Activity\Data->deleteActivities()
4. /var/www/html/nextcloud/apps/activity/lib/BackgroundJob/ExpireActivities.php line 54
   OCA\Activity\Data->expire()
5. /var/www/html/nextcloud/lib/public/BackgroundJob/Job.php line 81
   OCA\Activity\BackgroundJob\ExpireActivities->run()
6. /var/www/html/nextcloud/lib/public/BackgroundJob/TimedJob.php line 103
   OCP\BackgroundJob\Job->start()
7. /var/www/html/nextcloud/lib/public/BackgroundJob/TimedJob.php line 93
   OCP\BackgroundJob\TimedJob->start()
8. /var/www/html/nextcloud/cron.php line 152
   OCP\BackgroundJob\TimedJob->execute()
at 2023-11-19T11:10:44+02:00
```

It's strange, as I do not use Oracle.

### Steps to reproduce

1. Have a look at the logreader.
2. Observe the core error.

### Expected behavior

No errors

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by joshtrichards** [comment]
Duplicate of nextcloud/activity#1384

**Comment by joshtrichards** [comment]
Fix is already done - it'll be in 27.1.4
